From c2e8164fc29c57ccc2bea82dda9edab51abc8234 Mon Sep 17 00:00:00 2001
From: Elad Nachman <enachman@marvell.com>
Date: Tue, 25 Jan 2022 17:18:20 +0200
Subject: [PATCH 11/40] phylink: add SFP remove handler

Signed-off-by: Elad Nachman <enachman@marvell.com>
---
 drivers/net/phy/phylink.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/drivers/net/phy/phylink.c b/drivers/net/phy/phylink.c
index fef1416dcee4..2ad4dce717ee 100644
--- a/drivers/net/phy/phylink.c
+++ b/drivers/net/phy/phylink.c
@@ -2232,6 +2232,8 @@ static int phylink_sfp_config(struct phylink *pl, u8 mode,
 		return ret;
 	}
 
+	pl->link_config.an_enabled = phylink_test(pl->supported, Autoneg);
+
 	phylink_dbg(pl, "requesting link mode %s/%s with support %*pb\n",
 		    phylink_an_mode_str(mode), phy_modes(config.interface),
 		    __ETHTOOL_LINK_MODE_MASK_NBITS, support);
@@ -2287,7 +2289,14 @@ static int phylink_sfp_module_insert(void *upstream,
 
 	return phylink_sfp_config(pl, MLO_AN_INBAND, support, support);
 }
+static void phylink_sfp_module_remove (void *upstream)
+{
+    struct phylink *pl = upstream;
 
+    ASSERT_RTNL();
+
+    linkmode_zero(pl->supported);
+}
 static int phylink_sfp_module_start(void *upstream)
 {
 	struct phylink *pl = upstream;
@@ -2392,6 +2401,7 @@ static const struct sfp_upstream_ops sfp_phylink_ops = {
 	.attach = phylink_sfp_attach,
 	.detach = phylink_sfp_detach,
 	.module_insert = phylink_sfp_module_insert,
+	.module_remove = phylink_sfp_module_remove,
 	.module_start = phylink_sfp_module_start,
 	.module_stop = phylink_sfp_module_stop,
 	.link_up = phylink_sfp_link_up,
-- 
2.17.1

