From eff15471f1a5b4249ffe9c75e3a4fe6a7ed81367 Mon Sep 17 00:00:00 2001
From: Elad Nachman <enachman@marvell.com>
Date: Tue, 25 Jan 2022 17:27:17 +0200
Subject: [PATCH 13/40] Import Serhiy Boiko patch to Allow TC NAT to handle
 offloaded statistics

Signed-off-by: Elad Nachman <enachman@marvell.com>
---
 net/sched/act_nat.c | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/net/sched/act_nat.c b/net/sched/act_nat.c
index 7dd6b586ba7f..6d0e88d9df00 100644
--- a/net/sched/act_nat.c
+++ b/net/sched/act_nat.c
@@ -254,6 +254,16 @@ static int tcf_nat_act(struct sk_buff *skb, const struct tc_action *a,
 	return TC_ACT_SHOT;
 }
 
+static void tcf_nat_stats_update(struct tc_action *a, u64 bytes, u64 packets,
+				 u64 drops, u64 lastuse, bool hw)
+{
+	struct tcf_nat *nat = to_tcf_nat(a);
+	struct tcf_t *tm = &nat->tcf_tm;
+
+	tcf_action_update_stats(a, bytes, packets, drops, hw);
+	tm->lastuse = max_t(u64, tm->lastuse, lastuse);
+}
+
 static int tcf_nat_dump(struct sk_buff *skb, struct tc_action *a,
 			int bind, int ref)
 {
@@ -311,6 +321,7 @@ static struct tc_action_ops act_nat_ops = {
 	.id		=	TCA_ID_NAT,
 	.owner		=	THIS_MODULE,
 	.act		=	tcf_nat_act,
+	.stats_update	=	tcf_nat_stats_update,
 	.dump		=	tcf_nat_dump,
 	.init		=	tcf_nat_init,
 	.walk		=	tcf_nat_walker,
-- 
2.17.1

